### PR TITLE
CFE-676: Feature/decouple cert manager operator as a clusteroperator

### PR DESCRIPTION
**Pull request by thejasn**
Description
---
Decouples `cert-manager-operator` from registering as a clusteroperator. After this change the operator will not show up in the list of `ClusterOperators` on the console. 

Verified on cluster by installing bundle.
image

### CODE REVIEW

**Comment by thejasn** [comment]
Wait for rebase post #83 being merged.

File: pkg/controller/deployment/generic_deployment_controller.go [hunk 1]
```diff
@@ -22,6 +22,13 @@ import (
 	"github.com/openshift/cert-manager-operator/pkg/operator/operatorclient"
 )
 
+// TODO: This is just a temporary placeholder.
```
Comment (TrilokGeer) [inline pkg/controller/deployment/generic_deployment_controller.go:25]: In this PR, client registration is removed. Am I right to understand, the status can be seen in implementation of CFE-668?

Comment (thejasn) [reply pkg/controller/deployment/generic_deployment_controller.go:25]: Status of deployments (3 controllers from upstream) can be seen in CFE-668.

File: pkg/controller/deployment/cert_manager_controller_set.go [hunk 2]
```diff
@@ -24,7 +23,7 @@ type CertManagerControllerSet struct {
 func NewCertManagerControllerSet(
 	kubeClient kubernetes.Interface,
 	kubeInformersForTargetNamespace v1helpers.KubeInformersForNamespaces,
-	configClient configv1client.ConfigV1Interface,
+	// configClient configv1client.ConfigV1Interface,
```
Comment (swghosh) [inline pkg/controller/deployment/cert_manager_controller_set.go:26]: Nit: can we remove this line?

File: pkg/controller/deployment/generic_deployment_controller.go [hunk 3]
```diff
@@ -64,7 +70,7 @@ func newGenericDeploymentController(
 			kubeInformersForTargetNamespace.Core().V1().Namespaces().Informer(),
 		},
 		controller,
-		openshiftClusterConfigClient,
+		&dummyClusterConfigClient{},
```
Comment (swghosh) [inline pkg/controller/deployment/generic_deployment_controller.go:73]: IIUC, reading code from library-go seems like this config client is unused in terms of functionality within the controller, right?

[1] https://github.com/openshift/library-go/blob/master/pkg/operator/apiserver/controller/workload/workload.go

Comment (swghosh) [reply pkg/controller/deployment/generic_deployment_controller.go:73]: Correspondingly, then we could replace `&dummyClusterConfigClient{}` with `dummyClusterConfigClient{}`, if that works (by removing the additional pointer reference) without causing any type issues.

**Review by swghosh — COMMENTED** [review]
Mostly LGTM, just left a minor comments, which are mostly nits / questions seeking some context to myself.
For the time being this looks like a harmless change given it helps remove ClusterOperator references in the cluster.

File: pkg/controller/deployment/generic_deployment_controller.go (hunk 3 above)
Comment (thejasn) [reply pkg/controller/deployment/generic_deployment_controller.go:73]: This gets completely removed in #85 [38c59a](https://github.com/openshift/cert-manager-operator/pull/85/commits/38c59a46e59f400ee1af924e58606b3590ad54bf).

**Comment by thejasn** [comment]
/label docs-approved
Since this PR does not affect docs.

**Comment by thejasn** [comment]
/cc @davemulford 
This fixes UI/UX issue https://issues.redhat.com/browse/CM-35.

**Comment by thejasn** [comment]
Fixes CM-35

**Pull request merged**